src/analysis/retail/rogue/outlaw/modules/apl/AplCheck.tsx

```diff
@@ -0,0 +1,268 @@
+    )),
+  },
+  {
+    spell: TALENTS.BLADE_RUSH_TALENT,
+    condition: describe(energyCondition(60, 85), (tense) => (
+      <>you {tenseAlt(tense, 'are', 'were')} under ~70/80 energy</>
+    )),
+  },
+  {
+    spell: TALENTS.KILLING_SPREE_TALENT,
+    condition: describe(and(energyCondition(40, 60), notInStealthCondition()), (tense) => (
+      <>you {tenseAlt(tense, 'are', 'were')} under ~50/60 energy</>
+    )),
+  },
+  {
+    spell: TALENTS.ROLL_THE_BONES_TALENT,
+    condition: rtbCondition(),
+  },
+  {
+    spell: SPELLS.VANISH,
+    condition: and(
+      buffMissing(SPELLS.AUDACITY_TALENT_BUFF),
+      describe(
+        and(
+          buffStacks(SPELLS.OPPORTUNITY, { atMost: 3 }),
+          //This is a given no point displaying it
+          notInStealthCondition(),
+          //We want to allow the user to press vanish at max cp, but that is not a requirement
+          or(not(hasFinisherCondition()), optionalRule(hasFinisherCondition())),
+        ),
+        (tense) => (
+          <>
+            {' '}
+            you {tenseAlt(tense, 'have', 'had')} less than max stacks of{' '}
+            <SpellLink id={SPELLS.OPPORTUNITY} />
+          </>
+        ),
+      ),
+    ),
+  },
+  {
+    spell: TALENTS.SHADOW_DANCE_TALENT,
+    condition: describe(
+      and(
+        buffMissing(SPELLS.AUDACITY_TALENT_BUFF),
+        buffMissing(SPELLS.OPPORTUNITY),
+        //This is a given no point displaying it
+        notInStealthCondition(),
+        //We want to allow the user to press dance at max cp, but that is not a requirement
+        or(not(hasFinisherCondition()), optionalRule(hasFinisherCondition())),
+      ),
+      (tense) => (
+        <>
+          <SpellLink id={SPELLS.AUDACITY_TALENT_BUFF} /> and <SpellLink id={SPELLS.OPPORTUNITY} />{' '}
+          {tenseAlt(tense, 'are', 'were')} missing
+        </>
+      ),
+    ),
+  },
+];
+
+const COMMON_FINISHER: Rule[] = [
+  {
+    spell: SPELLS.BETWEEN_THE_EYES,
+    condition: and(
+      debuffMissing(SPELLS.BETWEEN_THE_EYES, {
+        timeRemaining: 4000,
+        //Since BtE as a variable duration depending on cp spent this is inacurate for now
+        duration: 19500,
+        pandemicCap: 1,
+      }),
+      describe(
+        and(
+          hasFinisherCondition(),
+          //We allow the user to not press BtE when in dance
+          or(
+            buffMissing(SPELLS.SHADOW_DANCE_BUFF),
+            optionalRule(buffPresent(SPELLS.SHADOW_DANCE_BUFF)),
+          ),
+        ),
+        (tense) => <>the finisher condition {tenseAlt(tense, 'is', 'was')} met</>,
+      ),
+    ),
+  },
+  {
+    spell: SPELLS.SLICE_AND_DICE,
+    condition: and(
+      hasFinisherCondition(),
+      buffMissing(SPELLS.SLICE_AND_DICE, {
+        timeRemaining: 18000,
+        //Since SnD as a variable duration depending on cp spent this is inacurate for now
+        duration: 45000,
+        pandemicCap: 1.3,
+      }),
```

Comment:
this probably needs to have a custom condition written for variable duration buffs, but i don't think that is a blocker.